modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java

```diff
@@ -1521,6 +1523,27 @@ public void deploysAppZipWithExtensionPlugin() throws Exception {
   }
 
   @Test
+  public void deploysAppZipWithExtensionXmlPlugin() throws Exception {
+    String moduleFileName = "module-bye.xml";
+    String extensionName = "bye-extension";
+    MuleArtifactDescriberBuilder builder =
+        new MuleArtifactDescriberBuilder().setName(extensionName).setMinMuleVersion(new MuleVersion("4.0.0"));
+    builder.withExtensionModelDescriber().setId(XmlBasedDescriber.DESCRIBER_ID).addProperty("resource-xml", moduleFileName);
+
+    final ArtifactPluginFileBuilder byeXmlExtensionPlugin = new ArtifactPluginFileBuilder(extensionName)
+        .containingResource("module-byeSource.xml", moduleFileName)
+        .configuredWith(builder.build());
+
+    ApplicationFileBuilder applicationFileBuilder = new ApplicationFileBuilder("appWithExtensionXmlPlugin")
+        .definedBy("app-with-extension-xml-plugin-config.xml").containingPlugin(byeXmlExtensionPlugin);
+    addPackedAppFromBuilder(applicationFileBuilder);
+
+    startDeployment();
+
+    assertDeploymentSuccess(applicationDeploymentListener, applicationFileBuilder.getId());
```

Comment:
talked offline, the concern Kraan is pointing out is validated in other test cases in this PR